### ISSUE TITLE
Unnecessary updates

### ISSUE THREAD

**Issue by antulik** [brendon/positioning#18]
Hi, thanks for the gem. 

It seems there are unnecessary updates when unrelated column is changed

---
Given model

```ruby
#  id                  :uuid             not null, primary key
#  fetching            :boolean          default(FALSE), not null
#  position            :integer          not null
#  type                :text             not null
#  project_id          :uuid             not null
#  index_segments_on_project_id_and_type_and_position  (project_id,type,position) UNIQUE
class Segment < ApplicationRecord
  self.inheritance_column = nil
  positioned on: [:project_id, :type]
end
```

When I run:
```ruby
Segment.last.update! fetching: true
```

This is the log
```
  Segment Load (0.9ms)  SELECT "segments".* FROM "segments" ORDER BY "segments"."id" DESC LIMIT $1  [["LIMIT", 1]]
  TRANSACTION (0.1ms)  BEGIN
  Segment Maximum (0.5ms)  SELECT MAX("segments"."position") FROM "segments" WHERE "segments"."project_id" = $1 AND "segments"."type" = $2  [["project_id", "0191029e-4502-7f7a-ad87-ea8293b6bcda"], ["type", "characterisation"]]
  Segment Exists? (0.3ms)  SELECT 1 AS one FROM "segments" WHERE "segments"."project_id" = $1 AND "segments"."type" = $2 AND "segments"."id" = $3 LIMIT $4  [["project_id", "0191029e-4502-7f7a-ad87-ea8293b6bcda"], ["type", "characterisation"], ["id", "0191029e-4608-715e-94ff-3ae4a89a46d0"], ["LIMIT", 1]]
  Segment Pluck (0.2ms)  SELECT "segments"."position" FROM "segments" WHERE "segments"."id" = $1 LIMIT $2  [["id", "0191029e-4608-715e-94ff-3ae4a89a46d0"], ["LIMIT", 1]]
  Segment Update All (0.8ms)  UPDATE "segments" SET "position" = $1 WHERE "segments"."id" = $2  [["position", 0], ["id", "0191029e-4608-715e-94ff-3ae4a89a46d0"]]
  Segment Update All (0.4ms)  UPDATE "segments" SET "position" = "position" * -1 WHERE "segments"."project_id" = $1 AND "segments"."type" = $2 AND "segments"."position" = $3  [["project_id", "0191029e-4502-7f7a-ad87-ea8293b6bcda"], ["type", "characterisation"], ["position", 1]]
  Segment Update All (0.5ms)  UPDATE "segments" SET "position" = "position" * -1 - 1 WHERE "segments"."project_id" = $1 AND "segments"."type" = $2 AND "segments"."position" <= $3  [["project_id", "0191029e-4502-7f7a-ad87-ea8293b6bcda"], ["type", "characterisation"], ["position", -1]]
  Segment Update (0.4ms)  UPDATE "segments" SET "updated_at" = $1, "position" = $2, "fetching" = $3 WHERE "segments"."id" = $4  [["updated_at", "2024-07-31 04:49:37.866613"], ["position", 1], ["fetching", true], ["id", "0191029e-4608-715e-94ff-3ae4a89a46d0"]]
  TRANSACTION (1.4ms)  COMMIT
```

Expected

There should be a single sql query to change `fetching` column and no changes to `position` column

Versions
positioning (0.2.3)

**Comment by brendon** [comment]
Hi @antulik, do you have a default value for your `position` column?

Here's the code that does the change detection. Let me know if you can dig a bit deeper to find the cause. Or you could perhaps try to create a failing test case:

https://github.com/brendon/positioning/blob/0c3be35b233744d68720c6acf65dcad9308c9b7f/lib/positioning/mechanisms.rb#L22-L39

Let me know if you need any assistance :)

**Comment by antulik** [comment]
Starts here:
https://github.com/brendon/positioning/blob/0c3be35b233744d68720c6acf65dcad9308c9b7f/lib/positioning/mechanisms.rb#L25

Sets the position here:
https://github.com/brendon/positioning/blob/0c3be35b233744d68720c6acf65dcad9308c9b7f/lib/positioning/mechanisms.rb#L123

which makes `position_changed?` to be `true`:
https://github.com/brendon/positioning/blob/0c3be35b233744d68720c6acf65dcad9308c9b7f/lib/positioning/mechanisms.rb#L27

which executes this:
https://github.com/brendon/positioning/blob/0c3be35b233744d68720c6acf65dcad9308c9b7f/lib/positioning/mechanisms.rb#L28

And executes this:
https://github.com/brendon/positioning/blob/0c3be35b233744d68720c6acf65dcad9308c9b7f/lib/positioning/mechanisms.rb#L36

---

I don't fully understand the logic, but seems like `solidify_position` shouldn't be called when `position` hasn't changed

**Comment by brendon** [comment]
Ok, on line 25 try changing it to `solidify_position if position_changed?` and see if that helps your situation. I've tried it on this end and the tests still pass so that could be a good optimisation.

**Comment by brendon** [comment]
I've released `0.2.4` that should fix this. I'll close this for now but let me know if there's still a problem.
